src/tribler-core/tribler_core/modules/libtorrent/restapi/downloads_endpoint.py

```diff
@@ -434,7 +434,7 @@ def get_files_info_json(self, download):
             num_files = len(download.tdef.get_files())
             if not all([0 <= index < num_files for index in selected_files_list]):
                 return RESTResponse({"error": "index out of range"}, status=HTTP_BAD_REQUEST)
-            download.set_selected_file_indexes(selected_files_list)
+            download.config.set_selected_file_indexes(selected_files_list)
```

Comment:
The names are a bit messed up right now (something I intend to address in https://github.com/Tribler/tribler/pull/5170). Are u sure this works? Maybe it should be `LibtorrentDownloadImpl.set_selected_files`?

Reply:
It definitely works. I've checked it by hand several times by including/excluding files through the GUI. Everything was fine.

The error was exactly that: `set_selected_file_indexes` was moved to `config` everywhere but in the test and the endpoint. The test did not catch it because of extensive mocking :man_shrugging:

Reply:
That's unexpected. And Tribler actually downloads the right files?

Reply:
Yes

Reply:
Well, at least it is consistent in the GUI.

Reply:
Well, you were right. After waiting 'till the download finished, the thing did not have any effect at all. It seems I was just lucky to select exactly those files from the test download (some music album) that were downloaded the most :dizzy_face:

Reply:
I'll finish this one after you're done with `LibtorrentDownloadImpl`